pkg/controller/builder.go

```diff
@@ -185,3 +195,60 @@ func replaceImageHook(placeholder string, desiredImageEnvVar string) deploymentc
 		return []byte(newDeployment), nil
 	}
 }
+
+func updateEnv(con *corev1.Container, name, value string) ([]corev1.EnvVar, error) {
+	for _, e := range con.Env {
+		if e.Name == name {
+			return con.Env, fmt.Errorf("unexpected environment variable %q=%q in container %q while building manifests", e.Name, e.Value, con.Name)
+		}
+	}
+	if value == "" {
+		return con.Env, nil
+	}
+	klog.FromContext(context.Background()).WithName("builder").V(1).Info("Updated environment", "container", con.Name, "key", name, "value", value)
+	return append(con.Env, corev1.EnvVar{Name: name, Value: value}), nil
+}
+
+func setProxyEnvs(con *corev1.Container, vars map[string]string) error {
+	var err error
+	// The order these are put into the Container matter for hashing
+	// The ordering of maps varies by design in golang, so we have to sort a slice of keys
+	keys := maps.Keys(vars)
+	slices.Sort(keys)
```

Comment:
Complete nit, take it or leave it: An alternative to sorting the keys is to have something like this in the loop, and then not passing things through a map at all:
```go
for _, c := range containers {
  for _, envVar := range []corev1.EnvVar{
    {HTTPSProxy, proxyConfig.Status.HTTPSProxy},
    {HTTPProxy,  proxyConfig.Status.HTTPProxy},
    {NoProxy,    proxyConfig.Status.NoProxy},
  } {
    if err := updateEnv(c, envVar); err != nil {
      errs = append(errs, err)
    }
  }
}
```

Reply:
Yeah, that might be cleaner. @perdasilva suggested the use of a map (which caused some headaches), but this might be better, avoids the whole sorting thing and what-not...